src/main/java/ch/njol/skript/SkriptEventHandler.java

```diff
@@ -137,13 +142,20 @@ private static void check(Event event, EventPriority priority) {
 
 		for (Trigger trigger : triggers) {
 			SkriptEvent triggerEvent = trigger.getEvent();
-			if (triggerEvent.getEventPriority() != priority || !triggerEvent.check(event))
+			if (triggerEvent.getEventPriority() != priority)
 				continue;
 
 			logTriggerStart(trigger);
 			Object timing = SkriptTimings.start(trigger.getDebugLabel());
-
-			trigger.execute(event);
+			if (!Bukkit.isPrimaryThread() && !trigger.getEvent().canExecuteAsynchronously()) {
+				if (Task.callSync(() ->!triggerEvent.check(event)))
+					continue;
+				Task.callSync(() -> trigger.execute(event));
```

Comment:
```suggestion
				Task.callSync(() -> {
					if (triggerEvent.check(event))
						trigger.execute(event);
					return null;
				});
```
Can't we just do this?

Reply:
No, a call sync is a callback not a consumer or runnable.

Reply:
Does returning null not cover that? We could also just return a boolean of whether the trigger ran? Is there any latency involved with the scheduler's callSyncMethod (e.g. ran on the next tick)?